wallet/src/Pos/Util/Mnemonic.hs

```diff
@@ -294,12 +286,20 @@ instance Buildable (SecureLog (Mnemonic mw)) where
     build _ =
         "<mnemonic>"
 
-instance Buildable MnemonicErr where
+instance Buildable Basement.String where
+    build = bprint . show . fromUtf8String
+
+instance Buildable (Checksum csz) where
+    build = bprint . show
+
+instance Buildable (MnemonicErr csz) where
     build = \case
-        MnemonicErrInvalidEntropyLength l ->
-            bprint ("Entropy must be a sequence of " % build % " bytes") l
-        MnemonicErrFailedToCreate ->
-            bprint "Invalid Mnemonic words"
+        MnemonicErrInvalidWord (ErrInvalidDictionaryWord w)->
+            bprint ("Invalid dictionary word: "%build%"") w
+        MnemonicErrInvalidEntropy (ErrInvalidEntropyLength e a) ->
+            bprint ("Invalid entropy length: got "%build%", expected "%build) e a
```

Comment:
We can give a unit here to make the message a bit clearer `got "%build%" bits, expect "%build%" bits")`